src/MongoDB.Driver.Core/Core/Connections/ConnectionInitializer.cs

```diff
@@ -92,59 +107,124 @@ public ConnectionDescription InitializeConnection(IConnection connection, Cancel
 
             return description;
         }
+        
+        internal BsonDocument AddAuthenticatorCustomizationsToIsMasterCommand(
```

Comment:
This method name is misleading because we do NOT always add the customizations...

Reply:
1. Simplest change: rename the method to "CustomizeIsMasterCommand"?

2. I could refactor it so that the method always adds the customizations (moving the if check outside of the method). That'd yield a tiny bit more code duplication. Strictly speaking an authenticator may choose not to customize the initial isMaster though: only a `DefaultAuthenticator` is guaranteed to change the initial isMaster.

3. A bigger change would be to tweak the interface so that the Authenticators return the requested customizations in a BsonDocument (possibly empty). We could get the customizations and merge them at this level.

Reply:
Method renamed, as discussed offline.